tensorflow_addons/optimizers/tests/lazy_adam_test.py

```diff
@@ -96,163 +96,122 @@ def _test_sparse(dtype):
+            tf.constant([1]),
+            tf.constant([2, 1]),
+        )
+        repeated_update_opt = lazy_adam.LazyAdam()
+        aggregated_update_opt = lazy_adam.LazyAdam()
+        for _ in range(3):
+            repeated_update_opt.apply_gradients(
+                [(grad_repeated_index, repeated_index_update_var)]
+            )
+            aggregated_update_opt.apply_gradients(
+                [(grad_aggregated, aggregated_update_var)]
+            )
+            np.testing.assert_allclose(
+                aggregated_update_var.numpy(), repeated_index_update_var.numpy()
+            )
+
+
+@pytest.mark.parametrize("use_callable_params", [True, False])
+@pytest.mark.parametrize("dtype", [tf.half, tf.float32, tf.float64])
+def test_basic(use_callable_params, dtype):
+    # Initialize tf for numpy implementation.
+    m0, v0, m1, v1 = 0.0, 0.0, 0.0, 0.0
+    var0_np = np.array([1.0, 2.0], dtype=dtype.as_numpy_dtype)
+    grads0_np = np.array([0.1, 0.1], dtype=dtype.as_numpy_dtype)
+    var1_np = np.array([3.0, 4.0], dtype=dtype.as_numpy_dtype)
+    grads1_np = np.array([0.01, 0.01], dtype=dtype.as_numpy_dtype)
+
+    var0 = tf.Variable(var0_np)
+    var1 = tf.Variable(var1_np)
+    grads0 = tf.constant(grads0_np)
+    grads1 = tf.constant(grads1_np)
+
+    def learning_rate():
+        return 0.001
+
+    if not use_callable_params:
+        learning_rate = learning_rate()
+
+    opt = lazy_adam.LazyAdam(learning_rate=learning_rate)
+
+    # Run 3 steps of Adam
+    for t in range(3):
+        beta_1_power, beta_2_power = get_beta_accumulators(opt, dtype)
+        test_utils.assert_allclose_according_to_type(0.9 ** (t + 1), beta_1_power)
+        test_utils.assert_allclose_according_to_type(0.999 ** (t + 1), beta_2_power)
+        opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
+
+        var0_np, m0, v0 = adam_update_numpy(var0_np, grads0_np, t, m0, v0)
+        var1_np, m1, v1 = adam_update_numpy(var1_np, grads1_np, t, m1, v1)
+
+        # Validate updated params
+        test_utils.assert_allclose_according_to_type(var0_np, var0.numpy())
+        test_utils.assert_allclose_according_to_type(var1_np, var1.numpy())
+
+
+@pytest.mark.parametrize("dtype", [tf.half, tf.float32, tf.float64])
+def test_tensor_learning_rate(dtype):
+    # Initialize tf for numpy implementation.
+    m0, v0, m1, v1 = 0.0, 0.0, 0.0, 0.0
+    var0_np = np.array([1.0, 2.0], dtype=dtype.as_numpy_dtype)
+    grads0_np = np.array([0.1, 0.1], dtype=dtype.as_numpy_dtype)
+    var1_np = np.array([3.0, 4.0], dtype=dtype.as_numpy_dtype)
+    grads1_np = np.array([0.01, 0.01], dtype=dtype.as_numpy_dtype)
+
+    var0 = tf.Variable(var0_np)
+    var1 = tf.Variable(var1_np)
+    grads0 = tf.constant(grads0_np)
+    grads1 = tf.constant(grads1_np)
+    opt = lazy_adam.LazyAdam(tf.constant(0.001))
+
+    # Fetch params to validate initial values
+    np.testing.assert_allclose([1.0, 2.0], var0.numpy())
+    np.testing.assert_allclose([3.0, 4.0], var1.numpy())
```

Comment:
Maybe using `assert_allclose_according_to_type` is more reasonable here?

Reply:
Sure thing

Reply:
Actually I don't even think we need those two lines anymore. Those checks have nothing to do with LazyAdam. Let me remove them.